catalog-rest-service/src/main/java/org/openmetadata/catalog/security/DefaultAuthorizer.java

```diff
@@ -60,7 +65,15 @@ public void init(AuthorizerConfiguration config, Jdbi dbi) {
     PolicyCache.initialize();
     RoleCache.initialize();
     LOG.debug("Admin users: {}", adminUsers);
-    initializeUsers();
+    // For a Basic Auth , an admin can during server startup decide who all can be admin by adding admin principal
+    // when a registration is done through registration api, then we add the user as admin or normal user
+    // checking whether that username was bootstrapped into server during startup as admin
+    if (!ConfigurationHolder.getInstance()
```

Comment:
@mohitdeuex any reason to change this. We can follow the same mechanism no?

Reply:
@harshach the problem is , currently we create the admin accounts based on admin principals, but in basic auth we have "password" which I don't want to take in yaml, because if there are multiple adminPrinicpal we probable need to sequentially accepts all admin user password, so what this does is we still take adminPrincipals, but we don't create a account for him on server startup , when the user registers for the OM like any other user, we see that if the username is one of adminPrincipal and we create his account as admin , this ensure that the flow is intact and on similar line with other users, and we still figure out who the admins are
Here is that check :-

    if (ConfigurationHolder.getInstance()
        .getConfig(ConfigurationHolder.ConfigurationType.AUTHORIZERCONFIG, AuthorizerConfiguration.class)
        .getAdminPrincipals()
        .contains(userName)) {
      newUser.setIsAdmin(true);
    }